src/main/java/seedu/binbash/command/DeleteCommand.java

```
@@ -30,16 +29,20 @@ public DeleteCommand(ItemList itemList, String keyword) {
         ));
     }
 
-    public boolean execute() {
+    public boolean execute(Ui ui, ItemList itemList) {
         if (isIndex) {
-            // Ensure index out of bounds error is caught by Parser.
+            if (index <= 0 || index > itemList.getItemCount()) {
+                executionUiOutput = "Index is out of bounds!";
+                return true;
+            }
```

I was thinking that we could probably throw an Exception here, instead of just returning an error message? Though this would mean that we have to put in a `try-catch` block in `BinBash.java` and print the exception message there.

---

You have a point. This is definitely up to whether you see an out of bounds index as severe enough for an exception.

---

True. I reckon that we could probably leave it as it is for now.